### ISSUE TITLE
reject does not interact well with OMS imported via URLs

### ISSUE THREAD

**Issue by tillmo** [spechub/Hets#1579]
When analysing the following with `hets -g`
```
logic OWL
ontology InputSpace1 =
  <http://ontohub.org/sandbox/inconsistent.omn>
  reject Ax1
end
```
I get a development graph with two nodes both having an empty theory (using the "show theory" menu).
By constrast, when analysing the same with the imported ontology just expanded:
```
logic OWL

ontology InputSpace1 =
  Prefix: : <http://www.omg.org/spec/DOL/DOL-terms/>
  Ontology: <http://www.omg.org/spec/DOL/DOL-terms/> 
  Class: A DisjointWith: B
  Class: B
  Class: C
  Individual: a Types: A and B
    reject Ax1
end
```
I get the expected OWL theories.

**Comment by tillmo** [comment]
this is rather urgent because it currently causes the monster paper workflow to fail

**Comment by mcodescu** [comment]
The library items of the first file are:
```
Logic_decl (LogicDescr (Logic_name "OWL" Nothing Nothing) Nothing nullRange) nullRange
logic OWL
Spec_defn InputSpace1 (Genericity (Params []) (Imported []) nullRange) (Annoted {item = Filtering (Annoted {item = Spec_inst <http://ontohub.org/sandbox/inconsistent.omn> [] Nothing nullRange, opt_pos = nullRange, l_annos = [], r_annos = []}) (FilterSymbolList False [SymbItems AnyEntity [QN {namePrefix = "", localPart = "Ax1", iriType = Abbreviated, expandedIRI = "", iriPos = nullRange}]] nullRange), opt_pos = nullRange, l_annos = [], r_annos = []}) nullRange
```
If the filtering ```reject Ax1``` is replaced with another structuring operation (I have tried hiding, because the generated Spec is similar to the one for filtering, and the same method, ```anaSpec``` is used for analysis of the specification we hide symbols of, and union with the empty specification followed by a reject, as well as the case when no structuring is involved and ```inputSpace1``` is just ```<http://ontohub.org/sandbox/inconsistent.omn>```), a ```Download_items``` for the URL is generated, and thus the analysis of a ```Spec_inst``` finds a specification for the URL in the global env, e.g.:
```
Analyzing file /home/mcodescu/repos/testSpecs/reject2.dol as library reject2
Download_items <http://ontohub.org/sandbox/inconsistent.omn> (UniqueItem <http://ontohub.org/sandbox/inconsistent.omn>) nullRange
Downloading <http://ontohub.org/sandbox/inconsistent.omn> ...
Analyzing library <http://www.omg.org/spec/DOL/DOL-terms/>
Logic_decl (LogicDescr (Logic_name "OWL" Nothing Nothing) Nothing nullRange) nullRange
logic OWL
Spec_defn <http://www.omg.org/spec/DOL/DOL-terms/> (Genericity (Params []) (Imported []) nullRange) (Annoted {item = Basic_spec OntologyDocument {prefixDeclaration = fromList [("","http://www.omg.org/spec/DOL/DOL-terms/"),("dc","http://purl.org/dc/elements/1.1/"),("owl","http://www.w3.org/2002/07/owl#"),("rdf","http://www.w3.org/1999/02/22-rdf-syntax-ns#"),("rdfs","http://www.w3.org/2000/01/rdf-schema#"),("xml","http://www.w3.org/XML/1998/namespace"),("xsd","http://www.w3.org/2001/XMLSchema#")], ontology = Ontology {name = QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/", iriType = Full, expandedIRI = "", iriPos = nullRange}, imports = [], ann = [[]], ontFrames = [Frame (ClassEntity (Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/A", iriType = Full, expandedIRI = "", iriPos = nullRange}))) [AnnFrameBit [] (AnnotationFrameBit Declaration)],Frame (ClassEntity (Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/B", iriType = Full, expandedIRI = "", iriPos = nullRange}))) [AnnFrameBit [] (AnnotationFrameBit Declaration)],Frame (SimpleEntity (Entity {label = Nothing, entityKind = NamedIndividual, cutIRI = QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/a", iriType = Full, expandedIRI = "", iriPos = nullRange}})) [AnnFrameBit [] (AnnotationFrameBit Declaration)],Frame (Misc []) [ListFrameBit (Just (EDRelation Disjoint)) (ExpressionBit [([],Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/A", iriType = Full, expandedIRI = "", iriPos = nullRange})),([],Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/B", iriType = Full, expandedIRI = "", iriPos = nullRange}))])],Frame (SimpleEntity (Entity {label = Nothing, entityKind = NamedIndividual, cutIRI = QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/a", iriType = Full, expandedIRI = "", iriPos = nullRange}})) [ListFrameBit (Just Types) (ExpressionBit [([],ObjectJunction IntersectionOf [Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/A", iriType = Full, expandedIRI = "", iriPos = nullRange}),Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/B", iriType = Full, expandedIRI = "", iriPos = nullRange})])])],Frame (ClassEntity (Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/A", iriType = Full, expandedIRI = "", iriPos = nullRange}))) [AnnFrameBit [] (AnnotationFrameBit Declaration)],Frame (ClassEntity (Expression (QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/B", iriType = Full, expandedIRI = "", iriPos = nullRange}))) [AnnFrameBit [] (AnnotationFrameBit Declaration)],Frame (SimpleEntity (Entity {label = Nothing, entityKind = NamedIndividual, cutIRI = QN {namePrefix = "http", localPart = "//www.omg.org/spec/DOL/DOL-terms/a", iriType = Full, expandedIRI = "", iriPos = nullRange}})) [AnnFrameBit [] (AnnotationFrameBit Declaration)]]}} nullRange, opt_pos = nullRange, l_annos = [], r_annos = []}) nullRange
Analyzing spec <http://www.omg.org/spec/DOL/DOL-terms/>
... loaded <http://ontohub.org/sandbox/inconsistent.omn>
Logic_decl (LogicDescr (Logic_name "OWL" Nothing Nothing) Nothing nullRange) nullRange
logic OWL
Spec_defn InputSpace1 (Genericity (Params []) (Imported []) nullRange) (Annoted {item = Spec_inst <http://ontohub.org/sandbox/inconsistent.omn> [] Nothing nullRange, opt_pos = nullRange, l_annos = [], r_annos = []}) nullRange
Analyzing spec reject2?InputSpace1
```

I could not find what generates this ```Download_items``` in the other cases than filtering.

**Comment by mcodescu** [comment]
The problem is that in Static/AnalysisLibrary.hs/anaStringAux, spNs is empty for the first example and contains the URL if the filtering is missing.

**Comment by mcodescu** [comment]
One line solver in Syntax/AS_Structured.der.hs, see commit on the branch 1576_...

**Comment by tillmo** [comment]
solved by 57de75c
